### PR TITLE
Hide incoming shared

### LINKED ISSUE #1351: Provide setting to hide shared items from the home gallery
Not everyone might want to see the incoming shared on their home timeline.

ente should provide an option (across all clients) to hide such shared entries from the home screen.

### PR DESCRIPTION
**Pull request by i-aiymen**
## Description
Added option in settings (General -> advanced) to hide/unhide the shared items in home gallery

Closes #1351

### CODE REVIEW

**Comment by i-aiymen** [comment]
@ashilkn

File: lib/ui/settings/advanced_settings_screen.dart [hunk 1]
```diff
@@ -119,6 +121,32 @@ class _AdvancedSettingsScreenState extends State<AdvancedSettingsScreen> {
...
+                            MenuItemWidget(
+                              captionedTextWidget: CaptionedTextWidget(
+                                title: S.of(context).hideSharedItems,
+                              ),
+                              menuItemColor: colorScheme.fillFaint,
+                              singleBorderRadius: 8,
+                              alignCaptionedTextToLeft: true,
+                              trailingWidget: ToggleSwitchWidget(
+                                value: () =>
+                                  LocalSettings.instance.getHideSharedItem(),
```
Comment (ashilkn) [inline lib/ui/settings/advanced_settings_screen.dart:133]: Can improve readability if this is changed to a getter function with name `hideSharedItems`

**Review by ashilkn — APPROVED** [review]

**Review by ashilkn — DISMISSED** [review]

File: lib/ui/home/home_gallery_widget.dart [hunk 2]
```diff
@@ -34,8 +35,20 @@ class HomeGalleryWidget extends StatelessWidget {
...
             CollectionsService.instance.archivedOrHiddenCollectionIds();
+        final Set<int> collectionsToHide;
+        if (LocalSettings.instance.hideSharedItems) {
+          final incomingSharedCollectionIDs = CollectionsService.instance
+              .getSharedCollections()
+              .incoming
+              .map((e) => e.id)
+              .toSet();
+          collectionsToHide =
+              archivedOrHiddenCollectionIDs.union(incomingSharedCollectionIDs);
```
Comment (ua741) [inline lib/ui/home/home_gallery_widget.dart:48]: We are using `archivedOrHiddenCollectionIds` at bunch of places. We can simplify this change by modifying the same method.

``` dart
Set<int> archivedOrHiddenCollectionIds() {
    final bool archiveIncomingCollections = LocalSettings.instance.hideSharedItems;
    final int ownerID = _config.getUserID()!;
    final Set<int> result = <int>{};
    for (final collection in _collectionIDToCollections.values) {
      if (collection.isHidden() ||
          collection.isArchived() ||
          collection.hasShareeArchived()) {
        result.add(collection.id);
        continue;
      }
      if (archiveIncomingCollections) {
        if (collection.owner?.id != ownerID) {
          result.add(collection.id);
        }
      }
    }
    return result;
  }
```
cc @ashilkn

File: lib/utils/local_settings.dart [hunk 3]
```diff
@@ -13,6 +13,7 @@ class LocalSettings {
   static final LocalSettings instance = LocalSettings._privateConstructor();
   static const kCollectionSortPref = "collection_sort_pref";
   static const kPhotoGridSize = "photo_grid_size";
+  static const kHideSharedItems = "hide_shared_items";
```
Comment (ua741) [inline lib/utils/local_settings.dart:16]: Internally, I think we should still call it `archive_shared_items`

Note:
Archived items are searchable. User can view them without any on-device auth.
Hidden items are supposed to be not searchable. User should perform on-device auth before they can view them.

File: lib/ui/home/home_gallery_widget.dart (hunk 2 above)
Comment (i-aiymen) [reply lib/ui/home/home_gallery_widget.dart:48]: @ua741 Should I Update it?

Comment (ashilkn) [reply lib/ui/home/home_gallery_widget.dart:48]: @ua741 Got it

**Comment by i-aiymen** [comment]
@ua741 @ashilkn

**Comment by ua741** [comment]
Can we also rename the option text to `Archive shared albums`? Apart from that, lgtm.

**Comment by i-aiymen** [comment]
> Can we also rename the option text to `Archive shared albums`? Apart from that, lgtm.

Done @ua741

File: lib/ui/settings/advanced_settings_screen.dart [hunk 4]
```diff
@@ -119,6 +121,32 @@ class _AdvancedSettingsScreenState extends State<AdvancedSettingsScreen> {
                             const SizedBox(
                               height: 24,
                             ),
+                            MenuItemWidget(
+                              captionedTextWidget: CaptionedTextWidget(
+                                title: S.of(context).archiveSharedAlbums,
```
Comment (ua741) [inline lib/ui/settings/advanced_settings_screen.dart:126]: There's no entry for this string in the following file? 
```
lib/l10n/intl_en.arb
```

Ideally, lib/generated/l10n.dart should generate the entry based on `intl_en.arb`

Comment (i-aiymen) [reply lib/ui/settings/advanced_settings_screen.dart:126]: I have added to lib/generated/l10n.dart

Comment (i-aiymen) [reply lib/ui/settings/advanced_settings_screen.dart:126]: Should I add it to intl_en.arb? @ua741

File: lib/generated/l10n.dart [hunk 5]
```diff
@@ -2950,6 +2950,16 @@ class S {
     );
   }
 
+  /// `Archive Shared Albums`
+  String get archiveSharedAlbums {
+    return Intl.message(
+      'Archive Shared Albums',
```
Comment (ashilkn) [inline lib/generated/l10n.dart:2956]: Can change this to "Archive shared albums"

File: lib/ui/settings/advanced_settings_screen.dart (hunk 4 above)
Comment (i-aiymen) [reply lib/ui/settings/advanced_settings_screen.dart:126]: Done

File: lib/generated/l10n.dart (hunk 5 above)
Comment (i-aiymen) [reply lib/generated/l10n.dart]: Done

**Review by ashilkn — APPROVED** [review]

**Pull request merged**